geopandas/geodataframe.py

```diff
@@ -1652,6 +1652,57 @@ def astype(self, dtype, copy=True, errors="raise", **kwargs):
         # do not return a GeoDataFrame
         return pd.DataFrame(df)
 
+    def convert_dtypes(
+        self,
+        infer_objects=True,
+        convert_string=True,
+        convert_integer=True,
+        convert_boolean=True,
+        convert_floating=True,
```

Comment:
I would use just `*args, **kwargs` here as that is future proof. In this case, we would need to make effort to keep the signature in sync with pandas. It also means that you can remove the creation of `kwargs` below and compat check for `convert_floating`.